src/Search/Gambit/GroupIdGambit.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+/*
+ * This file is part of Flarum.
+ *
+ * For detailed copyright and license information, please view the
+ * LICENSE file that was distributed with this source code.
+ */
+
+namespace FoF\UserDirectory\Search\Gambit;
+
+use Flarum\Group\Group;
+use Flarum\Group\GroupRepository;
+use Flarum\Search\AbstractRegexGambit;
+use Flarum\Search\AbstractSearch;
+use Flarum\User\Search\UserSearch;
+use LogicException;
+
+class GroupIdGambit extends AbstractRegexGambit
+{
+    /**
+     * {@inheritdoc}
+     */
+    protected $pattern = 'groupId:(.+)';
+
+    /**
+     * @var GroupRepository
+     */
+    protected $groups;
+
+    /**
+     * @param \Flarum\Group\GroupRepository $groups
+     */
+    public function __construct(GroupRepository $groups)
+    {
+        $this->groups = $groups;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function conditions(AbstractSearch $search, array $matches, $negate)
+    {
+        if (!$search instanceof UserSearch) {
+            throw new LogicException('This gambit can only be applied on a UserSearch');
+        }
+
+        $groupIds = $this->extractGroupIds($matches);
+
+        // TODO: Use a JOIN instead (and don't forget to remove the findByName() method again)
```

Comment:
Is that TODO relevant?

Reply:
Nope, it's the result of a copy-paste. I agree with this, but what do you think about https://github.com/FriendsOfFlarum/user-directory/issues/5#issuecomment-640102590? Also, should this be integrated into the group gambit (which might be better tbh), or as it's own gambit?